dags/hindawi/parser.py

```diff
@@ -0,0 +1,200 @@
+            ),
+            CustomExtractor(
+                destination="arxiv_eprints", extraction_function=self._get_arxiv
+            ),
+            TextExtractor(
+                destination="journal_year",
+                source="ns0:metadata/ns1:record/ns0:datafield/[@tag='773']/ns0:subfield/[@code='y']",
+                prefixes=self.prefixes,
+                extra_function=lambda x: x,
+            ),
+            CustomExtractor(
+                destination="copyright_statement",
+                extraction_function=self._get_copyright_statement,
+            ),
+            CustomExtractor(
+                destination="copyright_year",
+                extraction_function=self._get_copyright_year,
+            ),
+            CustomExtractor(
+                destination="license", extraction_function=self._get_license
+            ),
+        ]
+        super().__init__(extractors)
+
+    def _get_authors(self, article: ET.Element):
+        authors_all = article.findall(
+            "ns0:metadata/ns1:record/ns0:datafield/[@tag='100']", self.prefixes
+        )
+        authors_all + (
+            article.findall(
+                "ns0:metadata/ns1:record/ns0:datafield/[@tag='700']", self.prefixes
+            )
+        )
+        authors = []
+        for author in authors_all:
+            author_ = {
+                "raw_name": author.find(
+                    "ns0:subfield[@code='a']", self.prefixes
+                ).text.replace("\n", ""),
+                "affiliations": self._get_affiliations(author),
+            }
+            orcid = author.find("ns0:subfield[@code='j']", self.prefixes)
+            if orcid.text:
+                author_["orcid"] = orcid.text
+            authors.append(author_)
+
+        return authors
+
+    def _get_affiliations(self, author):
+        affiliations = [
+            affiliation.text.replace("\n", "")
+            for affiliation in author.findall("ns0:subfield[@code='u']", self.prefixes)
+        ]
+        parsed_affiliations = [
+            {
+                "value": affiliation,
+                "organization": (",").join(affiliation.split(",")[:-1]),
+                "country": affiliation.split(", ")[-1:][0],
+            }
+            for affiliation in affiliations
+        ]
+        return parsed_affiliations
+
+    def _get_arxiv(self, article: ET.Element):
+        arxiv = article.find(
+            "ns0:metadata/ns1:record/ns0:datafield/[@tag='037']/ns0:subfield/[@code='9']",
+            self.prefixes,
+        )
+        if arxiv is None:
+            self.logger.error("there is no Arxiv!")
+            return None
+        if arxiv.text.lower() == "arxiv":
+            arxiv_value = article.find(
+                "ns0:metadata/ns1:record/ns0:datafield/[@tag='037']/ns0:subfield/[@code='a']",
+                self.prefixes,
+            ).text.lower()
+            return [{"value": ("").join(arxiv_value.replace("arxiv:", "").split())}]
```

Comment:
hmm I don't understand this part of the code, why a tuple, and why hardcoded `arxiv:`?

Reply:
Tuple? It's a dict
Because it might have `arxiv` in the value, but we don't need it. We need just digits. For example:
```xml
	<ns0:subfield code="a">arXiv:1608.04627</ns0:subfield>
```
we need just 1608.04627

Reply:
Why not regex? and why `("").join` we don't need the tuple

Reply:
☝️

Reply:
I was just removing whitespaces, strip() works here perfectly